pkg/configure/configure.go

```diff
@@ -55,21 +55,56 @@ func ReconfigureHandler(k8sClient client.Client) func(w http.ResponseWriter, r *
 			return
 		}
 
+		var quay v1.QuayRegistry
+		nsn := types.NamespacedName{
+			Namespace: reconfigureRequest.Namespace,
+			Name:      reconfigureRequest.QuayRegistryName,
+		}
+		if err := k8sClient.Get(context.Background(), nsn, &quay); err != nil {
+			log.Error(
+				err,
+				"failed to fetch QuayRegistry",
+				"name",
+				reconfigureRequest.QuayRegistryName,
+				"namespace",
+				reconfigureRequest.Namespace,
+			)
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+
 		newSecret := createUpdatedSecret(reconfigureRequest)
-		if err = k8sClient.Create(context.Background(), &newSecret); err != nil {
-			log.Error(err, "failed to create new config bundle secret")
+
+		// set any possibly missing properties by overlaying the new secret properties
+		// on top of the original secret properties.
+		var oldSecret corev1.Secret
+		nsn.Name = quay.Spec.ConfigBundleSecret
+		if err := k8sClient.Get(context.Background(), nsn, &oldSecret); err != nil {
+			log.Error(
+				err,
+				"failed to fetch QuayRegistry config bundle",
+				"name",
+				quay.Spec.ConfigBundleSecret,
+				"namespace",
+				reconfigureRequest.Namespace,
+			)
 			http.Error(w, err.Error(), http.StatusInternalServerError)
 			return
 		}
 
-		log.Info("created new config secret for QuayRegistry: " + reconfigureRequest.Namespace + "/" + reconfigureRequest.QuayRegistryName)
+		for prop, value := range oldSecret.Data {
```

Comment:
It feels to me like this logic belongs in `createUpdatedSecret`, i.e. the signature becomes: 
```go
// createUpdatedSecret takes the reconfigureRequest and the oldSecret and coalesces them into a new secret.
func createUpdatedSecret(reconfigureRequest request, oldSecret corev1.Secret) corev1.Secret
```
It would also make it easier to write tests without needing to instantiate a client etc. Thoughts?

Reply:
I agree

Reply:
Updated, PTAL.